### ISSUE TITLE
broken by bleadperl v5.21.3-432-g8a00edd

### ISSUE THREAD

**Issue by tonycoz** [rjbs/Color-Scheme#3]
See https://rt.perl.org/Ticket/Display.html?id=122730 and http://www.cpantesters.org/cpan/report/ae344c56-34d3-11e4-9aac-71b9cc4edbc1

I didn't see a comment from you on the ticket, so I wasn't sure if you'd seen it.

**Comment by rjbs** [comment]
Thanks, I am aware. This is a module I mostly inherited, so to fix this bug I need to go figure out the intent of the code that broke, so it hasn't happened yet.

**Comment by rjbs** [comment]
I tried to have a go at this, but it doesn't fail on my laptop.  It has nvsize=8, but the report suggests it requires nvsize=16 to fail.  I've started a build on my linode.  The main problem may be figuring out what *should* happen.

**Comment by tonycoz** [comment]
On Wed, Mar 04, 2015 at 07:20:09PM -0800, Ricardo Signes wrote:
> I tried to have a go at this, but it doesn't fail on my laptop.  It has nvsize=8, but the report suggests it requires nvsize=16 to fail.  I've started a build on my linode.  The main problem may be figuring out what *should* happen.

That was the problem I had when I briefly tried to debug it.

Tony

**Comment by rjbs** [comment]
Baffled.

This is on my debug-noise branch.  I built a perl with uselongdouble.  Here's a bit of perl:

```perl
    my @rgb = map {
        min( 255, _round( $v - ( $v - $self->{"base_$_"} * $k ) * $s ) )
    } qw( red green blue );
    warn sprintf "@rgb [@RGB] <%g %g %g>\n", $v, $k, $s;
```

When I run it on nvsize=8:

```
0 255 179 [0 255 178.5] <255 1.5 1>
0 179 125 [0 178.5 124.95] <178.5 1.05 1>
```

When I run on nvsize=16:

```
0 255 179 [0 255 178.5] <255 1.5 1>
0 178 125 [0 178.5 124.95] <178.5 1.05 1>
```

So, on the "working" perl, 178.5 rounds to 179.  On the "failing" perl, 178.5 rounds to 178.  I tried several other rounding routines:  POSIX::round, sprintf.  I really don't know what's up.

I Devel::Peek-ed them.  On nvsize=8:

```
SV = PVNV(0x7ff552811c20) at 0x7ff552324348
  REFCNT = 1
  FLAGS = (NOK,pNOK)
  IV = 0
  NV = 178.5
  PV = 0
```

On nvsize=16:

```
SV = PVNV(0x7ff552811c20) at 0x7ff552324348
  REFCNT = 1
  FLAGS = (NOK,pNOK)
  IV = 0
  NV = 178.5
  PV = 0
```

?????

**Comment by rjbs** [comment]
The fix provided in #4 would have special cased just a few values, which would require, later, noticing future breakage and fixing it with more exceptions.  Instead, in a325c4d I have done what I suggested and implemented a routine for comparing two arrayrefs of rgb hex triples.  This fixes the test suites on 5.21.9-with-long-doubles on my laptop.  Released as 1.07.
